bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiRpc.java

```diff
@@ -268,9 +268,9 @@ public ShellyDeviceProfile getDeviceProfile(ThingTypeUID thingTypeUID, @Nullable
         // Pro 3EM has 3 meters
         // Pro 2 has 2 relays, but no meters
         // Mini PM has 1 meter, but no relay
-        if (THING_TYPE_SHELLYPRO2.equals(thingTypeUID)) {
-            profile.numMeters = 0;
-        } else if (THING_TYPE_SHELLYPRO3EM.equals(thingTypeUID)) {
+        if (THING_TYPE_CAP_NUM_METERS.containsKey(thingTypeUID)) {
+            profile.numMeters = getInteger(THING_TYPE_CAP_NUM_METERS.get(thingTypeUID));
```

Comment:
This will prevent two lookups and also simplify the code:
```suggestion
        Integer numMeters = THING_TYPE_CAP_NUM_METERS.get(thingTypeUID);
        if (numMeters != null) {
            profile.numMeters = numMeters;
```